pkg/apiserver/handlers.go

```diff
@@ -331,6 +331,9 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 
 	}
 
+	// For types that aren't root scoped, we default to the "default" namespace if none is specified for backwards compatibility
+	namespaceUnlessRootScoped := ""
```

Comment:
This is only used if set, and only gets set in the `else` block below... namespaces are a special case and handled in the `if` block, so this function is still behaving as expected.